### PR TITLE
Disable registry v1

### PR DESCRIPTION
**Pull request by mikesimos**
Change-type: minor
Signed-off-by: Michael Angelos Simos <michalis@balena.io>

### CODE REVIEW

File: src/lib/device-config.ts [hunk 1]
```diff
@@ -42,9 +44,10 @@ export const generateConfig = (
 	const userPromise = getUser(req);
 
 	// Devices running ResinOS >=1.2.1 are capable of using Registry v2, while earlier ones must use v1
-	const registryHost = resinSemver.gte(osVersion, '1.2.1')
-		? REGISTRY2_HOST
-		: REGISTRY_HOST;
+	if (resinSemver.lte(osVersion, '1.2.0')) {
+		throw new BadRequestError('Supervisor API url must be specified');
```
Comment (Page-) [inline src/lib/device-config.ts]: Did you copy/paste the error message by chance?

Comment (mikesimos) [reply src/lib/device-config.ts]: @Page- added a `Deprecated ResinOS version` message for review.

File: src/lib/device-config.ts [hunk 2]
```diff
@@ -42,9 +44,10 @@ export const generateConfig = (
 	const userPromise = getUser(req);
 
 	// Devices running ResinOS >=1.2.1 are capable of using Registry v2, while earlier ones must use v1
-	const registryHost = resinSemver.gte(osVersion, '1.2.1')
-		? REGISTRY2_HOST
-		: REGISTRY_HOST;
+	if (resinSemver.lte(osVersion, '1.2.0')) {
+		throw new BadRequestError('Deprecated ResinOS version');
```
Comment (Page-) [inline src/lib/device-config.ts]: `resin` should be `balena` and specifying the version range makes it easier for users to fix the error :)
```suggestion
		throw new BadRequestError('balenaOS versions <= 1.2.0 are no longer supported, please update');
```

**Comment by mikesimos** [comment]
@Page- thanks for approving dude. I bundled your balenaOS deprecation message update commit suggestion with my previous commit. (due to lint/style, and our check-commits format checking ). Cheers!

**Pull request merged**